RI 61 new tabs

kaaloo opened the pull request:
- refactor search results to include all matches and suggestions
- implement new result filters
- implement infinite scroll for search results
- update translations for new button labels
- fix issue with running update-icons

kaaloo committed:
chore(client):

kaaloo committed:
fix(client): :wrench: Fix issue with running update-icons script

kaaloo committed:
feat(client): :sparkles: Finish implementing results filters UI

kaaloo committed:
Revert "feat(client): :sparkles: Finish implementing results filters UI"

This reverts commit 7b0656d1999dd55e9db6c121b99238ff5549aa7d.

kaaloo committed:
feat(client): :sparkles: Finish implementing results filters UI

kaaloo committed:
chore: :construction_worker: Make sure update-icons runs when building the client too

kaaloo committed:
fix(client): :lipstick: Fix hover over sort options

kaaloo committed:
fix(client): :bug: Fix page width issue and add dispositif tab button label logic

kaaloo committed:
feat(client): :sparkles: Calculate and display online resource count

kaaloo committed:
feat(client): :construction: WIP new rules for displaying search results

kaaloo committed:
feat(client): :construction: Display cards in rows instead of a single line

kaaloo committed:
refactor(client):

kaaloo committed:
fix(client): :lipstick: Fix display sort filter alignment

kaaloo committed:
fix(client): :bug: Improve positioning

kaaloo committed:
chore(client): :recycle: Normalize class names and simplify output

kaaloo committed:
feat(client): :sparkles: Implemented tab based filtering in results

kaaloo committed:
fix(client): :lipstick: Cards should have the same height

kaaloo committed:
fix(client): :lipstick: Switch to fixed 4 column grid to avoid issues with placing items in last row

kaaloo committed:
feat(client): :sparkles: Render cards progressively into the dom

kaaloo committed:
fix(client):

kaaloo committed:
test(client): :white_check_mark: Update snapshots

kaaloo committed:
Merge branch 'dev' into luis/ri-61-new-tabs

ledjay commented on apps/client/src/components/Pages/recherche/ResultsFilter/ResultsFilter.module.css:
```diff
@@ -0,0 +1,126 @@
+.container {
+  display: flex;
+  position: relative;
+  height: 40px;
```
Peut-être plus un padding en rem qu'une height en px au cas où demain il faille changer la hauteur des éléments ?

matthieu-fesselier replied:
jusqu'ici j'utilisais un utility `unit`  (`height: u(4)`)

Margot fixe tous ses espacements sur des multiples de 4. Ça permet, comme pour une mesure en rem, de pouvoir changer rapidement le jour où on le souhaite

ledjay commented on apps/client/src/components/Pages/recherche/ResultsFilter/ResultsFilter.module.css:
```diff
@@ -0,0 +1,126 @@
+
+.grid {
+  display: flex;
+  position: absolute;
```
Pourquoi une position absolute ici ? A cause des marges du container ?
C'est un détail mais ça peut causer du layout shift et google n'aime pas trop ;)

matthieu-fesselier replied:
je suis d'accord avec ça, peut être un copié-collé de Figma ?

ledjay commented on apps/client/src/components/Pages/recherche/ResultsFilter/ResultsFilter.module.css:
```diff
@@ -0,0 +1,126 @@
+  width: max-content;
+
+  /* Style */
+  border-radius: 4px;
```
Peut-être passer les border-radius en variable css histoire de les rendre plus modulables ?

matthieu-fesselier replied:
Ils sont déjà dispos normalement en effet

ledjay commented on apps/client/src/components/Pages/recherche/SearchResults/SearchResults.tsx:
```diff
@@ -70,6 +62,42 @@ const SearchResults = (props: Props) => {
+  const boundingRect = ref?.current?.getBoundingClientRect();
+  useLayoutEffect(() => {
+    if (boundingRect) {
+      const count = _.floor((boundingRect.width + 24) / (282 + 24));
```
Peut-être passer en full css via des container queries pour se passer d'un layout effect et alléger le code ?

ledjay commented on apps/client/src/components/Pages/recherche/ResultsFilter/ResultsFilter.tsx:
```diff
@@ -75,30 +101,45 @@ const ResultsFilter = (props: Props) => {
+                    query.type === option.key && styles.tab_button_label_selected,
+                  )}
+                >
+                  {/* @ts-ignore */}
```
Compliqué à typer ici j'ai l'impression :)

ledjay reviewed (commented):
@kaaloo Quelques petites questions/suggestions ;)

ledjay commented on apps/client/src/components/Pages/recherche/ResultsFilter/ResultsFilter.tsx:
```diff
@@ -110,21 +151,19 @@ const ResultsFilter = (props: Props) => {
-                      <>{t(option.value)}</>
-                      {isSelected && <EVAIcon name="checkmark-outline" fill="white" size={20} />}
-                    </DropdownItem>
+                    <DropdownMenu.Item key={i} asChild>
```
Ici les menu items ne réagissent pas à la touche tab lorsque le menu est ouvert. Il y a peut-être quelque chose à creuser au niveau de Radix, je peux jeter un oeil si veux, je pense que ça vaut la peine niveau accessibilité ;)

ledjay reviewed (commented):
Oups j'avais oublié ce petit détail

matthieu-fesselier commented on apps/client/src/components/Pages/recherche/ResultsFilter/ResultsFilter.module.css:
```diff
@@ -0,0 +1,126 @@
+  color: var(--text-action-high-grey);
+  text-align: center;
+
+  /* 2.Corps de texte/MD - Texte standard/Desktop & Mobile - Medium */
```
On a des utilities aussi ici pour éviter de se répéter

matthieu-fesselier reviewed (commented):
OK avec les commentaires de Jérémie, et tout bon pour moi sinon :)

ledjay committed:
:recycle: Few adjustments on search grid and filter bar

- New TabItem and TaBar components to enhance reusability - Enhanced accessibility for dropdown menu (tabbing works!) - Responsive adjustments (with container queries) -  - tsignore fix

ledjay commented:
@kaaloo J'ai fait un petit commit qui reprend le principal de ce dont nous avions discuté.

J'ai pris le temps de splitter la tab bar et les tab items pour qu'ils puissent éventuellement être facilement ré-utilisés, ça te va ?

J'ai également vu 2-3 petites choses en + en fouillant dans le figma : 
- il faudrait donner du feed-back lorsque l'on change de filtre (un transition ou autre) parce que pour le moment algolia va tellement vite que dans certain cas cela donne l'impression qu'il ne s'est rien passé (ça rejoint peut-être aussi le point de Margot sur le fait d'afficher un "squelette" de card au chargement)
- il va manquer la nouvelle version de l'écran aucun résultat, je peux l'intégrer rapidement si tu veux ;)
<img width="452" alt="image" src="https://github.com/user-attachments/assets/0bb76595-201e-430d-84b8-dd5973b4070f">

ledjay commented:
@matthieu-fesselier j'ai dû revoir deux petites choses mineures sur les cards dispositif. J'avais besoin de leur passer une taille, j'ai donc ajouté une prop className facultative qui me permet d'override la taille juste sur la recherche. Et j'avais un gap qui me provoquait un bug de débordement sur la grille... j'ai dû fixer au plus simple :) Je ne pense pas que cela aura un impact majeur mais je veux bien que tu jettes un oeil au cas où.

matthieu-fesselier commented:
@ledjay j'ai jeté un oeil et ça me semble ok !

kaaloo commented:
> @kaaloo J'ai fait un petit commit qui reprend le principal de ce dont nous avions discuté.
> 
> J'ai pris le temps de splitter la tab bar et les tab items pour qu'ils puissent éventuellement être facilement ré-utilisés, ça te va ?
> 
> J'ai également vu 2-3 petites choses en + en fouillant dans le figma :
> 
> * il faudrait donner du feed-back lorsque l'on change de filtre (un transition ou autre) parce que pour le moment algolia va tellement vite que dans certain cas cela donne l'impression qu'il ne s'est rien passé (ça rejoint peut-être aussi le point de Margot sur le fait d'afficher un "squelette" de card au chargement)
> * il va manquer la nouvelle version de l'écran aucun résultat, je peux l'intégrer rapidement si tu veux ;)
> 
> <img alt="image" width="452" src="https://private-user-images.githubusercontent.com/3043571/366339464-0bb76595-201e-430d-84b8-dd5973b4070f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxMzA1NTMsIm5iZiI6MTcyNjEzMDI1MywicGF0aCI6Ii8zMDQzNTcxLzM2NjMzOTQ2NC0wYmI3NjU5NS0yMDFlLTQzMGQtODRiOC1kZDU5NzNiNDA3MGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTJUMDgzNzMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTVhNGIxOGM0Zjk2YmEyYmJhMTJmODdmMTVlNThhZDIzYWM2MzRkYzRjZjM0NDNjYTBkOTRjZmJkMTliYjlmNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.NpkCk3BacD9_S4VjVASIYXdU5uN_U_DOESdlxiE-3uA">

Merci @ledjay c'est super !  J'ai appris pas mal de choses en étudiant tes amélioriations avec l'aide de copilot 😁 .  Parfait pour la componentisation des Tabs.  Pour l'écran lorsqu'il n'y a pas de résultats, [c'est un autre ticket](https://linear.app/refugiesinfo/issue/RI-31/rr-recherche-avec-zero-resultat), on peut merger ce PR tel quel je pense.
